### DISCUSSION TITLE
Debugger 2.0 / Modal vs Popover / Filter button / Overall interaction feedback

### DISCUSSION THREAD

**Discussion by sang-yeo** [twilio-labs/paste#928] [Office hours] [answered]
Hello, 
We've defined customer problems and testing concepts with customers. 
Questions I have...
- Do we need to have a Filter button in order to filter the list based on design concept 1. 
- Modal vs Popover component usage - concept 1 and concept 2 
- Best  practice on the disclosure component
--Is it better to close or expand the disclosure component when a screen is busy?

Invision link and pw are in the signup sheet.

**Comment by serifluous** [comment, marked as answer]
> Do we need to have a Filter button in order to filter the list based on design concept 1.

It depends if you intend for the data in the graph and table to update as a customer changes the filters. For that decision, I'd recommend asking the engineers on your team what the performance of the page would be like if the filter is constantly doing a new query for each form field change (and of course consider what that experience might be like for someone using the filters).

> Modal vs Popover component usage - concept 1 and concept 2

What's your specific question about component usage here, and which screens are you referring to? The popover/tooltip on the graph will be in the style of whatever High Charts allows, but your usage of it seems pretty standard. I left a comment on the modal on "Concept 1 - Main Request Inspector", and I'm also not sure what triggers this modal.

> Best practice on the disclosure component
> --Is it better to close or expand the disclosure component when a screen is busy?

This'll depend on how important the info in the disclosure is relative to the other content on the page.

**Reply by sang-yeo** [reply]
> This'll depend on how important the info in the disclosure is relative to the other content on the page.

Yeah, I thought about that one. This depends on customers use case. This page solves two use cases and the info inside of the disclosure support one of use case.

**Comment by sang-yeo** [comment]
> What's your specific question about component usage here, and which screens are you referring to? The popover/tooltip on the graph will be in the style of whatever High Charts allows, but your usage of it seems pretty standard. I left a comment on the modal on "Concept 1 - Main Request Inspector", and I'm also not sure what triggers this modal.

I will work you through the steps tomorrow morning. 
During our study, customers told us they like to see the inspector request data other than modal because the modal makes them feel isolated. So my concept 2 uses the popover component but I'm not sure that's the correct usage of the component.
